src/lib/createStardustComponent.tsx

```diff
@@ -0,0 +1,36 @@
+import createComponentInternal from './createComponent'
+import * as React from 'react'
+import * as _ from 'lodash'
+import { ComponentSlotClasses } from '../themes/types'
+
+export interface RenderStardustResultConfig {
+  classes: ComponentSlotClasses
+  rtl: boolean
+}
+
+export type RenderStardustComponentCallback<P> = (
+  config: RenderStardustResultConfig,
+  props: P,
+) => any
```

Comment:
`React.ReactNode` as result type?

Reply:
also, didn't we agree to have only only one argument for render?
```typescript
render: (props: P & RenderStardustResultConfig) => React.ReactNode
```

Reply:
Not sure if we agreed on this, but from user perspective, if I want to pass all props to the thing I am rendering I would have to manually extract the things from this one object which are part of the props. Having two different properties, allows me to do this:
`<button {...props} />` I would say we can return object which contains {config, props}, but this seems cleaner to me. I am open for changes if we decide that the other proposal should be taken.